### DISCUSSION TITLE
Einsatz als reinen Batteriemonitor für ein JK-BMS

### DISCUSSION THREAD

**Discussion by SvenTej** [hoylabs/OpenDTU-OnBattery#1446] [Q&A] [answered]
Hallo zusammen,
ich würde gerne den (oder ist es das?) OpenDTU zum monitoren meines Akkus verwenden.

Hintergrund:
Ich habe eine funktionierende Victron Infrastruktur sowohl bei einer Insel, als auch beim AC gekoppelten Hausakku.

Nun möchte ich lediglich die Batteriedaten (Zellspannungen, Ströme usw.) im Homeassistant protokollieren.

Geht das ootb oder muss ich da was anpassen, weil ich ja keinen Wechselrichter usw. habe sondern nur das BMS via RS485?

Wäre für jeden Tipp dankbar.

Gruß
S.

**Comment by D3R-ST3FAN** [comment]
Schau doch mal hier : 
https://github.com/syssi/esphome-jk-bms oder für das JK Inverter BMS
https://github.com/txubelaxu/esphome-jk-bms

Oder per alternativ per
https://github.com/Louisvdw/dbus-serialbattery direct an ein Venus GX Gerät.
Dort kannst du dann die Daten ja z.B. via MQTT abgreifen und hättest auch noch den andern Victron kram im HA.

**Comment by SvenTej** [comment]
Hi Stefan,
Findest du das das openDTU Projekt unpassend?

Die Batterie an das GX zu bringen und das ganze durch das Victron System zu tunneln möchte ich ungern machen weil:
- über dbus-serialbattery man nicht an alle Daten rankommt
- ich bei der Insel das Thema habe, dass der Akku im Winter getauscht wird (da kommt ein Nassakku dran) und dann kommt das GX mächtig durcheinander

Von daher wäre mir das openDTU eben schon das angenehmere, vor allem weil ich an meinem BKW schon das "Normale" openDTU dran habe. 

Braucht das openDTU zwingend einen Wechselrichter?

S.

**Comment by D3R-ST3FAN** [comment]
OpenDTU braucht keinen Wechselrichter....
Unpassend? Naja viele Wege...
Primär ist OpenDTU ja für die HM WR gedacht.

Wenn es was spezifisch für die JKs gibt wie die beiden ESP-Home Projekte sogar mit schönem HA-Dashboard passt das vielleicht schon OOTB.

Ist am Ende Geschmackssache.

Je nachdem, wo du am Ende hin möchtest wäre evtl. sogar
https://github.com/shining-man/bsc_fw/ einen Blick wert.

Musst du am Ende für deinen Use-Case entscheiden.

**Comment by schlimmchen** [comment, marked as answer]
Du kannst gerne OpenDTU-OnBattery einsetzen, um dein JK BMS zu überwachen und in HomeAssistent einzubinden. In der Doku findest du Informationen zum Anschluss. Im nächsten Release wird sogar der Live View oben keine sinnlosen Karten mehr enthalten, wenn du keine Wechselrichter registrierst.

Es ist aber möglich, dass du ein paar Wert im HomeAssistent nicht siehst, die du dir wünscht, weil nicht alle Sensoren/Werte bei HomeAssistent registriert werden, siehe #670.

**Reply by SvenTej** [reply]
Danke euch beiden für die Infos.
Ich bin mit dem Fusion Board ja durchaus in der Lage, beides zu machen.
Da müsste ja sowohl ein ESPhome als auch ein openDTU drauf zum laufen gebracht werden.

Ist jetzt ein schönes Projekt für die nächsten Wochen

**Reply by SvenTej** [reply]
Gibt es zu dem openDTU-fusion board irgendwo Schaltpläne?
Ich habe leider nix gefunden. 

Wäre hilfreich um die PinConfig usw. anzupassen ohne viel Reverse Engineering im Code betreiben zu müssen

**Reply by schlimmchen** [reply]
Das ist doch alles [dokumentiert](https://github.com/markusdd/OpenDTUFusionDocs)^^

**Reply by SvenTej** [reply]
Hmmm....also ich bin Elektronikentwickler und vielleicht bin ich blind. Aber ich sehe keinen Schaltplan oder?
Ich weiß nicht, wie die ganzen Bauteile außerhalb der WROOM angeschlossen sind, oder habe ich was übersehen?

**Reply by spcqike** [reply]
Kein Schaltplan, aber eine Tabelle mit dem pinmapping. 

https://github.com/markusdd/OpenDTUFusionDocs/tree/main?tab=readme-ov-file#pin-mapping-of-the-esp32-s3-on-opendtu-fusion-v2x

als elektroentwickler kannst du ja vielleicht ein passenden, universellen Schaltplan machen und dem Projekt beisteuern (: 

Nachtrag:
Oder habe ich deine Frage falsch verstanden?

**Reply by stefan123t** [reply]
Das prinzipielle pin_mapping.json funktioniert. 

Die Schaltpläne für die v2 hatte @markusdd im Discord mal zur Diskussion gestellt. Sind mit denen der v1 im Kanal #fusion-referenece-board angepinnt. Ich weiß nicht was die Änderungen zwischen v2 und v2.01 sind / waren. Vielleicht kann er ja die Schaltpläne der drei Versionen als Grafik auf seinem Github Repo bereitstellen, da immer wieder mal Fragen kommen.

https://discord.com/channels/984173303147155506/1060347063281401876/1085704088370565161

@markusdd Deine KiCAD Files wären mE für die Weiterentwicklung zB einer Fusion ODOB PCB Pro durch @swingstate in #541 oder eines weiteren CAN HATs mit MCP2515 für die Huawei Netzteile hilfreich, wie in #1132 / #1144 durch @skippermeister, @Manos1966 und @AndreasBoehm besprochen.

Vielleicht wollen sich die Hardware Protagonisten ja Mal dazu austauschen ?

**Reply by schlimmchen** [reply]
Warum sollte Markus die KiCAD Originaldaten rausgeben? Schaltplan als PDF gerne. Aber das Design? Meiner Meinung nach keine gute Idee und nicht erforderlich.

Und auch der Ruf nach einem CAN HAT mit MCP2515 ist -- bitte sieh mir meine Direktheit nach -- Unsinn. Der Huawei AC Charger Stack wird stattdessen mittelfristig auch andere (den auf dem CAN/Iso HAT) CAN Controller unterstützen müssen.

**Reply by SvenTej** [reply]
Hmm vielleicht hab ich mich falsch ausgedrückt:
Ich habe nun schon 2 openDTU-fusion boards bestellt und hier liegen.

Wenn ich die nun nicht für das openDTU-Projekt verwenden möchte, sondern Stefans Anregung aufgreifen möchte, nicht openDTU sondern das https://github.com/txubelaxu/esphome-jk-bms zu verwenden hab ich das Problem, dass ich nicht weiß, was an dem WROOM angeschlossen ist und wie.

Das reine Pinmapping hilft hier nicht weiter und daraus einen Schaltplan zu machen...wie soll das gehen?

Ein Schaltplan per PDF wäre ultimativ hilfreich, um das o.g. Projekt auch auf einem openDTU-fusion zum Laufen zu bekommen.
Ich würd jetzt ungern das fusion zurücksenden, denn ich finde den Community/Appliance Gedanken schon sehr gut. 
Aber ohne besagte Infos wirds schwer.

**Reply by schlimmchen** [reply]
Sven, jetzt konzentrier' dich mal kurz: In der Doku zum Fusion-Board steht ganz klar wo welcher Pin des ESP32-S3 abzuzapfen ist.

Beispiel: Wenn du dir das Bild ganz oben in der [README](https://github.com/markusdd/OpenDTUFusionDocs) anschaust, ist links oben in diesem Bild der SDA Pin. Der ist "nur" so benannt, der ist vorgesehen zum Einsatz als I²C Datenpin, daher heißt der SDA. In der Tabelle unten siehst du, dass der SDA Pin des OpenDTU-Fusion-Boards "in Wahrheit" GPIO2 des ESP32-S3 ist.

So kannst du mit allen Pins verfahren. Welche Info fehlt dir jetzt noch?

Vielleicht noch jene, dass die GPIOs allesamt per Software für jede Funktion eingestellt und genutzt werden können?

Beachte übrigens, dass das Fusion ja schon einen RS485 transceiver hat, den du natürlich verwenden wollen würdest mit dem anderen Projekt.

> Ich würd jetzt ungern das fusion zurücksenden, denn ich finde den Community/Appliance Gedanken schon sehr gut.

Gute Einstellung :+1:

Allerdings, ganz ehrlich, ich hab mich jetzt erwischt wie ich meine Zeit damit verbringe jemanden zu helfen, was anderes als OpenDTU-OnBattery zu verwenden. Das ist das Gegenteil von dem, was mich motiviert, und hat hier auch keinen Platz. Wenn du mehr Hilfe brauchst dabei, das fremde Projekt zu nutzen, dann musst du dort jemanden um Hilfe bitten.

**Reply by SvenTej** [reply]
> Sven, jetzt konzentrier' dich mal kurz

Glaub mir ich BIN konzentriert.
Da brauch ich keinen, der mir sonen Spruch bring.

> Welche Info fehlt dir jetzt noch?

Auf dem fusion sind an dem WROOM diverse andere Bauteiile.
Wie ist die Spannungsversorgung aufgebaut....was machen die RF Chips.....Fragen über Fragen.

> Das ist das Gegenteil von dem, was mich motiviert, und hat hier auch keinen Platz.

Gut, hatte bisher eigentlich vor zu versuchen, mit dem fusion das andere Projekt zu testen, das mir hier vorgeschlagen wurde. 
Aber vor dem Hintergrund dass ich wiederum nicht das Risiko eingehen möchte und werde, bei Try&Error die Beschaltung des Boards rauszubekommen geht es zurück und ich weiche auf einen anderen ESP32 aus. 

S.

**Reply by markusdd** [reply]
a) verstehe ich das problem nicht. In der Boarddoku sind ALLE pins des WROOM moduls vermerkt. wirklich alle.
b) Auf dem silkscreen ist an jedem pin-header aufgedruckt was was ist.
c) An den RF-Modulen gibt es nichts herumzuwürgen, die RF-Front-End- und Antennenbeschaltungen sind gemäß Herstellervorgaben und EMV anforderungen augeführt. Alle Pins die zum WROOM gehen sind, wie schon gesagt, lückenlos dokumentiert.
d) Die Stromversorgung ist ein stinknormaler AMS1117 3.3V Linearregler, der wie auf 98% aller ESP32 Boards die ich je gesehen habe, aus 5V 3.3V macht. keine Magie involviert.

Also ich kann schlicht nicht nachvollziehen welche Infos dir fehlen.

Der Grund warum hier weder Layouts noch Schaltpläne rumfliegen, ist, dass das Board eben für OpenDTU ein verbreiteter Standard werden sollte, der 'einfach geht'. Das haben wir - würde ich behaupten - recht gut erreicht nach ein paar tausend Stück.
Dazu gehört aber eben auch, dass nicht jeder Hobbybastler der meint was von Elektronik zu verstehen hergeht und sich denkt: "Ja das kann ich ja auch mal bei JLCPCB bestellen".

a) ist das kostentechnisch sehr ineffizient und kostet am Ende mehr als die Community-Edition und b) die BOM dieses Boards umfasst mehr als 50 Bauteile, im Falle der RFs teils sehr speziell ausgewählte Kapazitäten und Induktoren wo man nicht mal eben irgendwas nehmen kann, was auf den ersten Blick den richtigen F/H Wert hat.

Das ist der fundamentale Unterschied: Während alle anderen Boards einfach fertige RF-Module einkaufen sind sie auf dem Fusion integriert. Deshalb packt es all diese Funktionalität auf 5.5x5.5cm . Und der Endkunde weiß, dass er hier Qualität kriegt wo kein 'cutting corners' betrieben wurde und jeweils die Hälfte der Antennenbeschaltung fehlt.

Am Ende ist also der Hauptgrund, dass wir das für uns behalten nicht, dass wir da auf unseren Lorbeeren sitzen bleiben wollen (wir verwalten und managen das alles neben unseren normalen Full-Time-Jobs), sondern weil wir vermeiden wollen, dass da dann irgendwelche Chinesen oder andere neunmalkluge losmarschieren und denken Abklatsche dieser Boards produzieren zu können, die dann die Standards nicht einhalten und dann fällt das auf uns zurück.

Für alles was Erweiterungen angeht - sprich die connections zum ESP32, haben wir wie gesagt alles aufgeschrieben, auch wo das Zeg hingeht wenn die jeweiligen Aufsteckboards benutzt werden. Es gibt auch ein riesiges Bild in der Doku, wo mit Pfeilen alle Komponentengruppen beschrieben sind.

Ich weiß offen gesagt nicht, was wir noch mehr tun sollen.

**Reply by schlimmchen** [reply]
> Da brauch ich keinen, der mir sonen Spruch bring.

Jaaaaa, das war wieder nicht "freundlich" von mir, aber es ist frustrierend, dass es sich so anfühlt als würdest du nicht begreifen wollen...

> Auf dem fusion sind an dem WROOM diverse andere Bauteiile.

Ignoriere die einfach. Aus meinem Beispiel mit SDA folgt: Der mit SDA beschritete Pin geht direkt an GPIO2 des ESP32-S3, fertig. Da ist nichts dran/dazwischen. Das ist auch mit den anderen freien Pins so, also allen die an den Pin-Headern verfügbar sind.

> und denken Abklatsche dieser Boards produzieren zu können

Das ist auch mein Gedanke.

> Das haben wir - würde ich behaupten - recht gut erreicht nach ein paar tausend Stück.

In der Tat :sunglasses: :muscle: 

> Ich weiß offen gesagt nicht, was wir noch mehr tun sollen.

Dich erinnern, dass zwar immer mal wieder jemand schimpft, aber die anderen 99% der Nutzer nicht, weil's einfach läuft :wink:

**Reply by stefan123t** [reply]
> > Vielleicht kann er ja die Schaltpläne der drei Versionen als Grafik auf seinem Github Repo bereitstellen.

> Schaltplan als PDF gerne.

Ja von mir aus auch PDF statt Grafik ;)

> > Deine KiCAD Files wären mE für die Weiterentwicklung zB einer Fusion ODOB PCB Pro durch swingstate

> Das ist der fundamentale Unterschied: Während alle anderen Boards einfach fertige RF-Module einkaufen sind sie auf dem Fusion integriert.

Ich dachte daher auch eher an eine Zusammenarbeit von markusdd und swingstate um ggf. die Features des PCB Pro auf einem Fusion OpenDTU-OnBattery Pro Board anzubieten oder ist das alles bereits auf einem Fusion mit CAN HAT vorhanden ?

Bzgl. MCP2515 hatte ich neulich mit Manos diskutiert und das Ergebnis war, dass wir zwei CAN Bus Adapter (Controller + Transceiver) brauchen weil die Huawei mit einer anderen Baud Rate kommunizieren. Ich wollte das Issue/die Discussion auch verlinken, finde es aber gerade nicht.

Bzgl. Nutzung von OpenDTU-OnBattery wollte Sven doch eigentlich ODOB nutzen man (der andere Stefan) hat ihm nur nahegelegt ein anderes Projekt zu probieren oder zu verwenden.

Daher seine Nachfrage nach einem Schaltplan.
Ich finde das nicht verwerflich, da ich auch eher ein visueller Charakter bin und mit dem Schaltplan so etwas besser nachvollziehen kann als mit einer Tabelle.

Auch finde ich für die Nachhaltigkeit OpenHardware Projekte toll, weil man ggf. auch mal ein Bauteil austauschen kann falls was kaputt gehen sollte. Ganz viele Gründe die mich an Ahoy, OpenDTU und Fusion, sowie natürlich auch ODOB begeistern.

**Reply by Sorbat** [reply]
> https://discord.com/channels/984173303147155506/1060347063281401876/1085704088370565161

Danke für den Link Stefan ❤️ das hilft mir sehr weiter 👍

**Comment by SvenTej** [comment]
> Ignoriere die einfach.

Aha....und woher weiß ich jetzt, wie ich die Bauteile die ich nicht verwende in dem anderen Projekt (ja ich weiß böse wenn man solche Fragen stellt) auf passiv schalte?
Pin floaten lassen ? log. HIGH ? log . LOW ?
Jetzt vielleicht klarer was der Hintergrund meiner Frage ist?

>  dass zwar immer mal wieder jemand schimpft, 

Und ich schimpfe nicht sondern hab lediglich nachgefragt.

Aber wie gesagt: ich gebe das Vorhaben auf, das openDTU-fusion board mit einem anderen Projekt einzusetzen.....

**Reply by markusdd** [reply]
Das ist schade, aber unnötig: Wie Bernhard schon richtig sagte sind alle pins die in der Doku auf 'freely mappable' stehen ohne jede weitere beschaltung, mit denen aknnst du tun was du willst.
Wenn an Pins Pull-Ups/Down hängen steht das auch in der Doku, bei den LEDs stehen sogar die seriellen Widerstandswerte drin.
Alles was fast an Bauteile verdrahtet ist verrät dir auch die Doku woran es geht. Wenn du jetzt also Dinge 'abschalten' etc. möchtest reicht eins blick ins datenblatt des jeweiligen Bauteils, an welchen Pin es geht sagt dir die Doku.
